omnixai/explainers/ranking/agnostic/validity.py

```diff
@@ -0,0 +1,215 @@
+#
+# Copyright (c) 2022 salesforce.com, inc.
+# All rights reserved.
+# SPDX-License-Identifier: BSD-3-Clause
+# For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+#
+"""
+Ranking Explainer for tabular data.
+"""
+import numpy as np
+from typing import Callable, List
+from ...base import ExplainerBase
+from ....data.tabular import Tabular
+import scipy
+import itertools
+import pandas as pd
+
+from ....explanations.ranking.agnostic.validity import ValidExplanation
+
+
+class ValidityRankingExplainer(ExplainerBase):
+    """
+    Ranking Explainer for Tabular Data.
+    """
+
+    explanation_type = "local"
+    alias = ["rank"]
+
+    def __init__(
+        self,
+        training_data: Tabular,
+        features: List,
+        predict_function: Callable,
+        preprocessing_function: Callable = None,
+    ):
+        """
+        :param training_data: The data used to initialize a Ranking explainer. ``training_data``
+            can be the training dataset for training the machine learning model.
+        :param features: The list of features to be explained by the valid per-query algorithm
+        :param predict_function: The prediction function corresponding to the model to explain.
+            the outputs of the ``predict_function`` are the document scores.
+        :param preprocessing_function: The preprocessing function to process/transform the feature attributes
+        before prediction. The output of the preprocessing function is the input to the ranking model.
+        """
+        super().__init__()
```

Comment:
Better to add some assertions about input parameters, e.g., training_data should be a Tabular instance, and the values in "features" or "ignored_features" should be in training_data.columns.